exporter/trace/stackdriver/trace.go

```diff
@@ -68,9 +68,7 @@ func (e *traceExporter) ExportSpans(ctx context.Context, sds []*export.SpanData)
 		pbSpans[i] = protoFromSpanData(sd, e.projectID)
 	}
 	var cancel func()
-	if ctx == nil {
-		ctx, cancel = newContextWithTimeout(e.o.Context, e.o.Timeout)
-	}
+	ctx, cancel = newContextWithTimeout(ctx, e.o.Timeout)
```

Comment:
There are `ctx == nil` checks all over this code, could you clarify the intention? Testing for a nil context anywhere has a funny smell. I would like to see all the nil context checks removed.

Also, should you add a test for the case where context reaches the timeout and is canceled?

Reply:
I think ctx should not be nil here and the cancel() will always be nil. I'll check it with timeout tests.